external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -292,7 +293,7 @@ impl BuildConfig {
         if modified || install_dir_set {
             // (1) Write the Move.lock file if the existing one is `modified`, or
             // (2) `install_dir` is set explicitly, which may be a different directory, and where a Move.lock does not exist yet.
-            let lock = dependency_graph.write_to_lock(install_dir)?;
+            let lock = dependency_graph.write_to_lock(install_dir, Some(lock_path))?;
             if let Some(lock_path) = &self.lock_file {
                 lock.commit(lock_path)?;
             }
```

Comment:
We now pass a lock file path to `write_to_lock` (if it exists, the content is preserved module dependency graph info).

Side note: If lock files don't exist yet, this call is the one that currently initializes them. Ultimately, I want to factor out the responsibility of initializing lock files (with version and manifest digest) so that it's not in dependency graph code--it should be done wrt relevant commands (`sui move build`, `sui move publish`, ...)